test/tests/functional/pbs_reservations.py

```diff
@@ -173,3 +173,36 @@ def test_not_honoring_resvs(self):
 
         self.server.expect(JOB, {'job_state': 'Q'}, id=j1id)
         self.server.expect(JOB, {'job_state': 'Q'}, id=j2id)
+
+    def test_sched_cycle_starts_on_resv_end(self):
+        """
+        This test checks whether the sched cycle gets started
+        when the advance reservation ends.
+        """
+        now = int(time.time())
+        a = {'Resource_List.select': "1:ncpus=2",
+             'reserve_start': now + 5,
+             'reserve_end': now + 12,
+             }
+        r = Reservation(TEST_USER, a)
+        rid = self.server.submit(r)
+        print rid.split('.')[0]
+        attr = {'Resource_List.walltime': '00:00:20'}
+        j = Job(TEST_USER, attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'Q'},
+                           id=jid)
+        msg = "Job would conflict with reservation or top job"
+        self.server.expect(JOB, {ATTR_comment: msg}, op=MATCH_RE,
```

Comment:
Isn't this supposed to be (ATTR_comment: (MATCH_RE, msg)} ?

Reply:
This is fine. 
Following expect code lets you to to do this way. Check the else part in the following code snippet:

        for k, v in attrib.items():
            args = None
            if isinstance(v, tuple):
                operator = v[0]
                if len(v) > 2:
                    args = v[2:]
                val = v[1]
            else:
                operator = op

Reply:
While this will work, I think it'd be better to do a straight match against the actual comment.  The actual comment is "Not Running: Job would conflict with reservation or top job".  You won't need to do a regex match then.

Reply:
Done